src/freertos_drivers/esp32/Esp32WiFiManager.cxx

```diff
@@ -48,17 +48,42 @@
 #include <mdns.h>
 #include <tcpip_adapter.h>
 
-// ESP-IDF v4+ has a slightly different directory structure to previous
-// versions.
-#ifdef ESP_IDF_VERSION_MAJOR
-// ESP-IDF v4+
+// Starting in ESP-IDF v4.0 a few header files have been relocated so we need
+// to adjust the include paths accordingly. If the __has_include preprocessor
+// directive is defined we can use it to find the appropriate header files.
+// If it is not usable then we will default the older header filenames.
+#if defined __has_include
```

Comment:
Isn't the syntax for this `#if defined(__has_include)` ?

Reply:
It appears to work either way, updated it as you have posted above though.